app/models/comment.rb

```diff
@@ -0,0 +1,16 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  private
+
+  after_save :increment_commentscounter
+  before_destroy :decrement_commentscounter
+  def increment_commentscounter
+    author.increment!(:commentscounter)
+  end
+
+  def decrement_commentscounter
+    author.decrement!(:commentscounter)
+  end
```

Comment:
[OPTIONAL] Nice work so far. However, in order for the rails active callback method `after_save` to work effectively, you need to put it above the `private` declaration (that will serve as a trigger to call the other private methods) and put the `increment_commentscounter` and `decrement_commentscounter` methods under private.
It could be similar to the code below: 

              class Comment < ApplicationRecord
                   belongs_to :author, class_name: 'User'
                    belongs_to :post

                   after_save :increment_commentscounter
 
                   private

                   def increment_commentscounter
                        author.increment!(:commentscounter)
                   end

                   def decrement_commentscounter
                         author.decrement!(:commentscounter)
                   end
               end